File: 卡码/代码随想录/Go/哈希/赎金信/LeetCode_0383.go
//383. 赎金信
package main

func canConstruct(ransomNote string, magazine string) bool {
	hash:=make([]int,26)
	//解密字符串长度不够一定为false
	if len(ransomNote)>len(magazine) {
		return false
	}
	//先将解密字符串所有字符统计到数组里
	for _,ch:=range magazine {
		hash[ch-'a']++
	}
	for _,ch:=range ransomNote {
		hash[ch-'a']--
		//如果小于0说明原字符串有的解密字符串没有
		if hash[ch-'a']<0 {
			return false
		}
	}
	return true
}

File: 卡码/代码随想录/1/03-哈希表/Js/06-赎金信/LeetCode_0383.js
//383. 赎金信
const canConstruct = function(ransomNote, magazine) {
  if(ransomNote.length>magazine.length) return false;
  let hash=new Array(26).fill(0);
  for(let i=0;i<magazine.length;i++)
    hash[magazine.charCodeAt(i)-'a'.charCodeAt(0)]++;
  for(let i=0;i<ransomNote.length;i++){
    let ch=ransomNote.charCodeAt(i)-'a'.charCodeAt(0);
    hash[ch]--;
    if(hash[ch]<0) return false;
  }
  return true;
};
